crates/bevy_pbr/src/render/shadow_sampling.wgsl

```diff
@@ -0,0 +1,123 @@
+#define_import_path bevy_pbr::shadow_sampling
+
+#import bevy_pbr::mesh_view_bindings as view_bindings
+#import bevy_pbr::utils PI
+
+fn sample_shadow_map_hardware(light_local: vec2<f32>, depth: f32, array_index: i32) -> f32 {
+    // Do the lookup, using HW 2x2 PCF and comparison
+    return textureSampleCompareLevel(
+        view_bindings::directional_shadow_textures,
+        view_bindings::directional_shadow_textures_sampler,
+        light_local,
+        array_index,
+        depth,
+    );
```

Comment:
I didn't mean to remove the 'NO ARRAY TEXTURES' bit, just to make both cases `textureSampleCompareLevel`. Sorry if that was unclear. The 'NO ARRAY TEXTURES' stuff is needed for WebGL2/GLES 3.0 I think.